modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java

```diff
@@ -378,7 +381,41 @@ else if (consentType == CookiesConstants.CONSENT_TYPE_PERSONALIZATION) {
 		}
 
 		if (Validator.isNull(consentCookieValue)) {
-			return true;
+			try {
```

Comment:
Hey @oliverkecskemety we might want to use `CookiesConfigurationProvider` here. The logic to decide whether we get system or group config does not depend on the existence of a group but also that the site itself has defined a virtual host domain and user accesses the site using it.

Otherwise, cookies can be mixed up across sites if we use group-level config when accessing the site using the virtual instance hostname. Please take a look to https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2164228128/LPS-165962+Research+cases+for+cross-site+navigations+under+the+same+domain+in+terms+of+cookie+handling for more details.

If that helps we can have a call to discuss this case